docs/source/conf.py

```diff
@@ -301,6 +301,8 @@ def setup(app):
 # packages that cannot be installed in RTD
 autodoc_mock_imports = DIRAC_DOC_MOCK_LIST
 
+# (Optional) Prevent documenting inherited Pydantic internals
+autodoc_default_options = {"inherited-members": False}
```

Comment:
Something like

```suggestion
autodoc_default_options = {"inherited-members": "pydantic.main.BaseModel, object"}
```

Otherwise you are disabling all inherited-members, even those in DIRAC itself.

https://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html#directive-option-autoclass-inherited-members

> The members of classes listed in the argument to :inherited-members: are excluded from the automatic documentation. This defaults to [object](https://docs.python.org/3/library/functions.html#object) if no argument is provided, meaning that members of the object class are not documented. To include these, use None as the argument.

Reply:
There are other issues now, including the one I was trying to fix

```
/home/docs/checkouts/readthedocs.org/user_builds/dirac/checkouts/8336/src/DIRAC/DataManagementSystem/private/HttpStorageAccessHandler.py:docstring of http.server.BaseHTTPRequestHandler.send_error:6: WARNING: Block quote ends without a blank line; unexpected unindent. [docutils]
/home/docs/checkouts/readthedocs.org/user_builds/dirac/checkouts/8336/src/DIRAC/DataManagementSystem/private/HttpStorageAccessHandler.py:docstring of http.server.BaseHTTPRequestHandler.send_error:7: WARNING: Inline emphasis start-string without end-string. [docutils]
/home/docs/checkouts/readthedocs.org/user_builds/dirac/checkouts/8336/src/DIRAC/FrameworkSystem/private/standardLogging/LogLevels.py:docstring of DIRAC.FrameworkSystem.private.standardLogging.LogLevels.LogLevel.from_bytes:9: WARNING: Inline interpreted text or phrase reference start-string without end-string. [docutils]
/home/docs/checkouts/readthedocs.org/user_builds/dirac/checkouts/8336/src/DIRAC/FrameworkSystem/private/standardLogging/LogLevels.py:docstring of DIRAC.FrameworkSystem.private.standardLogging.LogLevels.LogLevel.to_bytes:8: WARNING: Inline interpreted text or phrase reference start-string without end-string. [docutils]
/home/docs/checkouts/readthedocs.org/user_builds/dirac/checkouts/8336/src/DIRAC/WorkloadManagementSystem/Utilities/JobModel.py:docstring of pydantic.main.BaseModel.model_json_schema:8: WARNING: Bullet list ends without a blank line; unexpected unindent. [docutils]
/home/docs/checkouts/readthedocs.org/user_builds/dirac/checkouts/8336/src/DIRAC/WorkloadManagementSystem/Utilities/JobModel.py:docstring of pydantic.main.BaseModel.model_json_schema:8: WARNING: Bullet list ends without a blank line; unexpected unindent. [docutils]
```

Reply:
Yes, this overwrites the settings here

https://github.com/DIRACGrid/DIRAC/blob/385b299ad56225ca94e344ce4384c9b92fbd9d5d/docs/docs.conf#L30-L33

Reply:
I pushed something that I hope fixes it: https://github.com/DIRACGrid/DIRAC/pull/8336/commits/3ad7d2da1dc4363dc6462091a1340857eb5a3bd9

Reply:
It does not fix it

Reply:
I am trying a different option

Reply:
Just FYI
```diff
(fix_docs_no_autodoc_inherited_pydantic *)$ git diff
diff --git a/docs/diracdoctools/cmd/codeReference.py b/docs/diracdoctools/cmd/codeReference.py
index 65a1deacb8..17c9f8074f 100644
--- a/docs/diracdoctools/cmd/codeReference.py
+++ b/docs/diracdoctools/cmd/codeReference.py
@@ -153,7 +153,7 @@ class CodeReference:
         if buildtype == "full":
             lines.append("   :members:")
             if not any(x in self.config.code_noInherited for x in [classname, fullclassname]):
-                lines.append("   :inherited-members:")
+                lines.append("   :inherited-members: BaseModel")
             lines.append("   :undoc-members:")
             lines.append("   :show-inheritance:")
             if classname in self.config.code_privateMembers:
diff --git a/src/DIRAC/FrameworkSystem/scripts/dirac_logout.py b/src/DIRAC/FrameworkSystem/scripts/dirac_logout.py
index bff243e251..0bb3aed6be 100644

```

This fixes it for me. `pydantic.BaseModel` or `pydantic.main.BaseModel` had no effect